engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -1889,12 +1898,22 @@ struct demon_hunter_sigil_t : public demon_hunter_spell_t
   {
     demon_hunter_spell_t::execute();
 
-    // TOCHECK -- Appears to have a 100ms ICD in the spell data
     if ( hit_any_target && p()->talent.demon_hunter.soul_sigils->ok() )
     {
       unsigned num_souls = as<unsigned>( p()->talent.demon_hunter.soul_sigils->effectN( 1 ).base_value() );
       p()->spawn_soul_fragment( soul_fragment::LESSER, num_souls, false);
     }
+    if ( hit_any_target && p()->talent.vengeance.cycle_of_binding->ok() )
+    {
+      std::vector<cooldown_t*> sigils_on_cooldown;
+      range::copy_if( sigil_cooldowns, std::back_inserter( sigils_on_cooldown ),
+                      []( cooldown_t* c ) { return c->down(); } );
+      if ( !sigils_on_cooldown.empty() )
+      {
+        cooldown_t* sigil_cooldown = sigils_on_cooldown[ rng().range( sigils_on_cooldown.size() ) ];
```

Comment:
Minor thing, but I think this may throw a warning unless you static_cast<int> the RNG result here.